build/commands/lib/config.js

```diff
@@ -539,6 +539,16 @@ Config.prototype.buildArgs = function () {
     // https://github.com/brave/brave-browser/issues/10334
     args.dcheck_always_on = this.isComponentBuild()
 
+    if (!this.isBraveReleaseBuild() && this.isComponentBuild()) {
```

Comment:
a Brave release build will never be a component build so this seems unnecessary

Reply:
@bridiver I did so in the event that we may decide one day to make Debug builds on CI via PR builder, if the PR builder also doesn't set this then sure I can remove it